# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/AntiTrackingIPCUtils.h
# Commit: 3b1dbe909aaf
# Full Hash: 3b1dbe909aaf1982c72db1e5bc8cb35208ae6c77
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-04-23 21:43:09
# Regressor Bug: 1616775
# File Overlap Count: 1
# Description:
#   Bug 1616775 - P4. Add CompleteAllowAccessFor IPDL r=timhuang,baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D71013
# ==============================================================================

diff -r 123b6c3a0ede -r 3b1dbe909aaf toolkit/components/antitracking/AntiTrackingIPCUtils.h
--- a/toolkit/components/antitracking/AntiTrackingIPCUtils.h	Thu Apr 23 14:28:41 2020 +0000
+++ b/toolkit/components/antitracking/AntiTrackingIPCUtils.h	Thu Apr 23 14:25:16 2020 +0000
@@ -10,6 +10,7 @@
 #include "ipc/IPCMessageUtils.h"
 
 #include "mozilla/ContentBlockingNotifier.h"
+#include "mozilla/ContentBlocking.h"
 
 namespace IPC {
 
@@ -24,6 +25,14 @@
           mozilla::ContentBlockingNotifier::StorageAccessGrantedReason::
               eOpener> {};
 
+// ContentBlocking::StorageAccessPromptChoices over IPC.
+template <>
+struct ParamTraits<mozilla::ContentBlocking::StorageAccessPromptChoices>
+    : public ContiguousEnumSerializerInclusive<
+          mozilla::ContentBlocking::StorageAccessPromptChoices,
+          mozilla::ContentBlocking::StorageAccessPromptChoices::eAllow,
+          mozilla::ContentBlocking::StorageAccessPromptChoices::
+              eAllowAutoGrant> {};
 }  // namespace IPC
 
 #endif  // mozilla_antitrackingipcutils_h